docs/extending.rst

```diff
@@ -270,13 +270,14 @@ Normally, unspecified field names are ignored by the validator. If you would lik
 
         @validates_schema(pass_original=True)
         def check_unknown_fields(self, data, original_data):
-            for key in original_data:
-                if key not in self.fields:
-                    raise ValidationError('Unknown field name {}'.format(key))
+            unknown = set(original_data) - set(self.fields)
+            if unknown:
+                errors = list(map("Unknown field '{}'".format, unknown))
+                raise marshmallow.ValidationError(errors)
```

Comment:
How about using ValidationError structure to point (machine-readable) to fields that have error?!
```python
raise marshmallow.ValidationError('Unknown field', list(unknown))
```
The reported result would be in a dictionary form like this:
```python
{'unknown_field1': 'Unknown field', 'unknown_field2': 'Unknown field'}
```